Explorer/Controller/CreateLogViewController.swift

```diff
@@ -152,9 +163,53 @@ class CreateLogViewController: UIViewController {
     @objc func dismissInputView() {
         self.view.endEditing(true)
         
+        dismissPickerView()
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    @objc func keyboardWillShow(notification: Notification) {
+        dismissPickerView()
+        
+        guard let userInfo = notification.userInfo else { return }
+        guard let keyboardSize = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue else { return }
+        let keyboardFrame = keyboardSize.cgRectValue
+        inputOffset = view.frame.height - (keyboardFrame.height + 18)
+        
+        var keyboardFocus: CGFloat = 0.0
+        
+        if createLogView.diveNumberTextField.isFirstResponder {
+            keyboardFocus = createLogView.diveNumberTextField.frame.origin.y
+        }
+
+        if createLogView.locationTextField.isFirstResponder {
+            keyboardFocus = createLogView.locationTextField.frame.origin.y
+        }
+
+        if createLogView.notesTextView.isFirstResponder {
+            keyboardFocus = createLogView.notesTextView.frame.origin.y + 36
```

Comment:
What does the `36` represent? Try to avoid constants like this. One better solution is to put it in a named variable so you at least know why it's there.

Another solution though, and maybe better, is to actually calculate the height of the `noteTextView` and adjust for that. What if you change the height of `notesTextView` in the future? You'll also need to update the `36` value as well.

These thoughts aren't that big of a deal in this scenario, but it's just something i think wanted to open up conversation on for future reference.

Reply:
Replaced with the textView height